File: app/graphql/mutations/respond_to_invite.rb
# frozen_string_literal: true

module Mutations
  class RespondToInvite < ::GraphQL::Schema::Mutation
    null false
    argument :responses, [InputTypes::Invite], required: true
    field :success, GraphQL::Types::Boolean, null: false

    def resolve(responses:)
      responses.each do |response|
        invite = Invite.find(response.invite_id)
        invite.update(status: response.status)
      end
      { success: true }
    end
  end
end

File: client/src/queries/respondToInvite.js
import gql from 'graphql-tag';
import { useMutation } from '@apollo/react-hooks';

const respondToInvite	= gql`
  mutation respondToInvite($invites: [Invite!]!){
     respondToInvite(responses: $invites){
      success
    }
  }
`;

export default respondToInvite;
